nodes/text/viewer_text_mk3.py

```diff
@@ -0,0 +1,226 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import bpy
+from bpy.props import StringProperty, BoolProperty, IntProperty
+
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import levelsOflist, changable_sockets, multi_socket, updateNode
+
+
+
+class SverchokViewerMK1(bpy.types.Operator):
+    """Sverchok viewerMK1"""
+    bl_idname = "node.sverchok_viewer_buttonmk1"
+    bl_label = "Sverchok viewer.mk1"
+    bl_options = {'REGISTER', 'UNDO'}
+
+    nodename = StringProperty(name='nodename')
+    treename = StringProperty(name='treename')
+    lines = IntProperty(name='lines', description='lines count for operate on',default=1000)
+
+    def execute(self, context):
+        node = bpy.data.node_groups[self.treename].nodes[self.nodename]
+        inputs = node.inputs
+
+        # outputs
+        outs  = ''
+        for insert in inputs:
+            if insert.is_linked:
+                label = insert.other.node.label
+                if label:
+                    label = '; node ' + label.upper()
+                name = insert.name.upper()
+                # vertices socket
+                if insert.other.bl_idname == 'VerticesSocket':
+                    itype = '\n\nSocket ' + name + label + '; type VERTICES: \n'
+
+                # edges/faces socket
+                elif insert.other.bl_idname == 'StringsSocket':
+                    itype = '\n\nSocket ' + name + label + '; type EDGES/POLYGONS: \n'
```

Comment:
Huh. @zeffii this is nearly the same as we were talking in #2058: text viewer nodes assumes that StringsSockets is for edges or polygons, but in fact it is for anything for now. @nortikin maybe put a bit more general phrase here, smth like "EDGES/POLYGONS/OTHER" ?

Also I'm not sure about caps :)